File: ui/lib/Models/Room.dart

import 'Point.dart';
import 'Users/User.dart';

class Room {
  final String id;
  final Point point;
  final User? assignedPatient;
  final int roomNumber;
  
  Room({
    required this.id,
    required this.point,
    required this.assignedPatient,
    required this.roomNumber,
  });
  
  factory Room.fromJson(Map<String, dynamic> json) {
    return Room(
      id: json['id'],
      point: Point.fromJson(json['point']),
      assignedPatient: json['assignedPatient'] != null ? User.fromJson(json['assignedPatient']) : null,
      roomNumber: json['roomNumber'],
    );
  }
}

File: Domain/Room.cs
namespace Domain;

public class Room
{
    public Guid Id { get; set; }
    public Point Point { get; set; }
    public User? AssignedPatient { get; set; }
    public int RoomNumber { get; set; }
    
    public Room(){}
    public Room(Point point, User user, int roomNumber)
    {
        Point = point;
        AssignedPatient = user;
        RoomNumber = roomNumber;
    }
}
